src/Pact/Native.hs

```diff
@@ -385,6 +387,56 @@ fromNamespacePactValue :: Namespace PactValue -> Namespace (Term Name)
 fromNamespacePactValue (Namespace n userg adming) =
   Namespace n (fromGuardPactValue userg) (fromGuardPactValue adming)
 
+dnUserGuard :: FieldKey
+dnUserGuard = "user-guard"
+
+dnAdminGuard :: FieldKey
+dnAdminGuard = "admin-guard"
+
+dnNamespaceName :: FieldKey
+dnNamespaceName = "namespace-name"
+
+describeNamespaceSchema :: NativeDef
+describeNamespaceSchema = defSchema "described-namespace"
+  "Schema type for data returned from 'describe-namespace'."
+  [ (dnUserGuard, tTyGuard Nothing)
+  , (dnAdminGuard, tTyGuard Nothing)
+  , (dnNamespaceName, tTyString)
+  ]
+
+describeNamespaceDef :: NativeDef
+describeNamespaceDef = defGasRNative "describe-namespace" describeNamespace
```

Comment:
Should this not be top-level only? Is this data supposed to be used for anything in particular, like deploying keysets or something?

I'm actually curious if this should almost be local only, or do we need to be able to use this data in say module deploys?

Reply:
Setting to toplevel  cuz i can't think of a use case for non-toplevel